### PR TITLE
Define and use $OF_DEMO_UID for demo mode

### PR DESCRIPTION
**Pull request by reedy**
Ref https://github.com/jpatokal/openflights/issues/1232

### CODE REVIEW

File: php/map.php [hunk 1]
```diff
@@ -31,7 +31,7 @@
 $challenge = $_SESSION["challenge"] ?? null;
 if (!$uid || empty($uid)) {
     // If not logged in, default to demo mode and warn app that we're (no longer?) logged in
-    $uid = 1;
+    $uid = $_SESSION["opentrid"];
```
Comment (chrisrosset) [inline php/map.php]: What does a trip ID have to do with the user ID?

File: php/stats.php [hunk 2]
```diff
@@ -8,7 +8,7 @@
 $public = "O"; // by default...
 
 // If not logged in, default to demo mode
-$uid = $_SESSION["uid"] ?? 1;
+$uid = $_SESSION["uid"] ?? $_SESSION["opentrid"];
```
Comment (chrisrosset) [inline php/stats.php]: Same as above, expected `$OF_DEMO_UID` here.

File: php/map.php (hunk 1 above)
Comment (reedy) [reply php/map.php]: I'm guessing I copied the wrong thing and wasn't paying any attention...

File: php/stats.php (hunk 2 above)
Comment (reedy) [reply php/stats.php]: Same here

File: php/flights.php [hunk 3]
```diff
@@ -43,17 +43,14 @@
...
+            // Naughty naughty, back to demo mode
+            $uid = $OF_DEMO_UID;
         }
         // No limit, do nothing
     } else {
         // Nope, default to demo mode
-        $uid = 1;
+        $uid = $OF_DEMO_UID;
     }
 }
```
Comment (reedy) [inline php/flights.php:56]: Some refactoring here would be nice to reduce the duplication a bit... But job for another day

**Comment by reedy** [comment]
@jpatokal You'll probably want to pre-emptively set `$OF_DEMO_UID = 1;` in `config.php` in production before this change is deployed...

**Pull request merged**